pxr/imaging/hdx/taskController.cpp

```diff
@@ -1930,26 +1950,63 @@ void
 HdxTaskController::SetColorCorrectionParams(
     HdxColorCorrectionTaskParams const& params)
 {
-    if (_colorCorrectionTaskId.IsEmpty()) {
-        return;
-    }
+    if (!_colorCorrectionTaskId.IsEmpty()) {
+        HdxColorCorrectionTaskParams oldParams = _delegate.GetParameter<
+            HdxColorCorrectionTaskParams>(_colorCorrectionTaskId,
+            HdTokens->params);
 
-    HdxColorCorrectionTaskParams oldParams = 
-        _delegate.GetParameter<HdxColorCorrectionTaskParams>(
-            _colorCorrectionTaskId, HdTokens->params);
+        // We assume the caller for SetColorCorrectionParams wants to set the
+        // OCIO settings, but does not want to override the AOV used to do color-
+        // correction on. (Currently this AOV is controlled via TaskController)
+        HdxColorCorrectionTaskParams newParams = params;
+        newParams.aovName = oldParams.aovName;
 
-    // We assume the caller for SetColorCorrectionParams wants to set the
-    // OCIO settings, but does not want to override the AOV used to do color-
-    // correction on. (Currently this AOV is controlled via TaskController)
-    HdxColorCorrectionTaskParams newParams = params;
-    newParams.aovName = oldParams.aovName;
+        if (newParams != oldParams) {
+            _delegate.SetParameter(_colorCorrectionTaskId, HdTokens->params,
+                newParams);
 
-    if (newParams != oldParams) {
-        _delegate.SetParameter(
-            _colorCorrectionTaskId, HdTokens->params, newParams);
+            GetRenderIndex()->GetChangeTracker().MarkTaskDirty(
+                _colorCorrectionTaskId, HdChangeTracker::DirtyParams);
+        }
 
-        GetRenderIndex()->GetChangeTracker().MarkTaskDirty(
-            _colorCorrectionTaskId, HdChangeTracker::DirtyParams);
+        if (!_presentTaskId.IsEmpty()) {
+            auto presentParams = _delegate.GetParameter<HdxPresentTaskParams>(
+                    _presentTaskId, HdTokens->params);
+            if (const auto windowDst = std::get_if<HgiWindowPresentParams>(
+                &presentParams.destination)) {
+
+                if (params.colorCorrectionMode.IsEmpty() ||
```

Comment:
It would be preferable that we track the Hgi target render color space, and the HgiWindow color space at a higher level, rather than hard coding  the values here, but I realize that such a change should be out of scope since there is more work to be done to ensure that Hgi works in the right color space throughout the rest of the code.

Reply:
I expect there'll be the flexibility to pick a rendering space as well ultimately, since it's true that different groups have different needs. Thanks for the thoughts, definitely OOS.